models/models.news.js

```diff
@@ -59,14 +59,44 @@ exports.fetchUsers = () => {
   });
 };
 
-exports.fetchArticles = () => {
-  return db
-    .query(
-      `SELECT articles.*,count(comments.article_id)::INT as comment_count FROM articles left JOIN comments ON comments.article_id=articles.article_id GROUP BY articles.article_id ORDER BY created_at DESC;`
-    )
-    .then(({ rows }) => {
+exports.fetchArticles = (sort_by = "created_at", order = "DESC", filter_by) => {
+  const validSorts = [
+    "article_id",
+    "comment_count",
+    "title",
+    "author",
+    "created_at",
+    "votes",
+  ];
+  const order_by = order.toUpperCase();
+  const validOrder = ["ASC", "DESC"];
+
+  let queryValue = [];
+  let queryString = `SELECT articles.*,COUNT(comments.article_id)::INT AS comment_count FROM articles 
+  LEFT JOIN comments ON comments.article_id=articles.article_id
+  GROUP BY articles.article_id
+  ORDER BY ${sort_by} ${order_by};`;
+
+  if (filter_by) {
+    queryValue.push(filter_by);
+    queryString = `SELECT articles.article_id,articles.title,articles.author,articles.votes,articles.topic,articles.created_at,COUNT(comments.article_id)::INT AS comment_count FROM articles left JOIN comments ON comments.article_id=articles.article_id WHERE topic=$1 GROUP BY articles.article_id  ORDER BY ${sort_by} ${order}`;
```

Comment:
This could be a little cleaner if you just added to the query string rather than reassigning it but its totally fine and works :)